python/ray/serve/_private/client.py

```diff
@@ -107,27 +107,31 @@ def shutdown(self) -> None:
             del self.handle_cache[k]
 
         if ray.is_initialized() and not self._shutdown:
-            ray.get(self._controller.shutdown.remote())
-            self._wait_for_deployments_shutdown()
-
-            ray.kill(self._controller, no_restart=True)
-
-            # Wait for the named actor entry gets removed as well.
-            started = time.time()
-            while True:
-                try:
-                    ray.get_actor(self._controller_name, namespace=SERVE_NAMESPACE)
-                    if time.time() - started > 5:
-                        logger.warning(
-                            "Waited 5s for Serve to shutdown gracefully but "
-                            "the controller is still not cleaned up. "
-                            "You can ignore this warning if you are shutting "
-                            "down the Ray cluster."
-                        )
-                        break
-                except ValueError:  # actor name is removed
-                    break
+            try:
+                ray.get(self._controller.set_shutting_down_flag.remote())
+                logger.warning(
+                    "`shutting_down` flag set on the controller. The controller "
```

Comment:
`shutting_down` flag is an implementation detail, so I don't think it would make much sense to the user.

```suggestion
                    "Serve controller received shutdown request. It "
```

Reply:
This should not be a `WARNING`-level log because it is part of expected operation.